lib/dor/item.rb

```diff
@@ -93,10 +93,9 @@ def self.add_workflow_for_item(druid)
     def self.create_workflow(druid)
       LyberCore::Log.debug "...adding workflow #{Dor::Config.release.workflow_name} for #{druid}"
 
-      # initiate workflow
+      # initiate workflow by making workflow service call
       with_retries(max_tries: Dor::Config.release.max_tries, base_sleep_seconds: Dor::Config.release.base_sleep_seconds, max_sleep_seconds: Dor::Config.release.max_sleep_seconds) do |_attempt|
-        obj = Dor.find(druid)
-        obj.create_workflow(Dor::Config.release.workflow_name)
+        Dor::Config.workflow.client.create_workflow('dor', druid, Dor::Config.release.workflow_name, Dor::WorkflowObject.initial_workflow(Dor::Config.release.workflow_name), {})
```

Comment:
This line was previously accomplished by https://github.com/sul-dlss/dor-services/blob/39eb6012a7b93fe2b0059d90410e0fc7f8c33813/lib/dor/models/concerns/processable.rb#L219

Can you explain why `{ :create_ds => create_ds, :lane_id => default_workflow_lane }` isn't necessary anymore and can just be `{}`?

Reply:
i was about to ask the same thing!

Reply:
We can just accept the defaults provided by the workflow service gem: https://github.com/sul-dlss/dor-workflow-service/blob/845edfe4160a165d62b06530138f71e5c816a5c9/lib/dor/services/workflow_service.rb#L44-L53

Reply:
Can we though?  Previously the 1st and 4th params were derived from the `name`, like:
```ruby
Dor::Config.workflow.client.create_workflow(
  Dor::WorkflowObject.initial_repo(name), 
  pid, 
  name, 
  Dor::WorkflowObject.initial_workflow(name), 
  opts
)
```
Notably, both `Dor::WorkflowObject.initial_repo(name)` and `Dor::WorkflowObject.initial_workflow(name)` make Solr lookups, like:
```ruby
:q=>"_query_:\"{!raw f=has_model_ssim}info:fedora/afmodel:Dor_WorkflowObject\" AND workflow_name_ssim:xyz"...}
```

Doing that twice in one line (underneath dor-services) is certainly a source of performance degradation, which is probably what prompted the ugly class-level caching [here](https://github.com/sul-dlss/dor-services/blob/39eb6012a7b93fe2b0059d90410e0fc7f8c33813/lib/dor/models/workflow_object.rb#L5-L6), since the object being retrieved is **the same one**.  

In any case, why can `item-release` assume that `Dor::WorkflowObject.initial_repo(name)` is always `'dor'`?  Especially if that value was coming from an indexed field in Solr?  That value's construction comes from [`WorkflowObjectDs`](https://github.com/sul-dlss/dor-services/blob/39eb6012a7b93fe2b0059d90410e0fc7f8c33813/lib/dor/datastreams/workflow_definition_ds.rb#L55) which relies on the (fetched) datastream XML.  To vouch for it always being 'dor' would mean knowing how every WF description DS XML was generated, today and in the future.  Here are all the [25 current objects in Solr](https://sul-solr-a.stanford.edu/solr/argo3_prod/select?q=has_model_ssim:%22info:fedora/afmodel:Dor_WorkflowObject%22&rows=100).

Reply:
Here's a gist of the objects w/ [their indexed workflowDefinition](https://gist.github.com/atz/674961f870f1f39ca357372a5e150279), (crudely unescaped).  The only value that seems like it could represent this `initial_repo` is `objOwnerId` and only 4 of 25 have `"DOR"`.

Reply:
I guess the old code always did pass `Dor::Config.release.workflow_name` every time, but what I don't get is why it could have presumed that to be correct.

Reply:
I think you are right about the first parameter, safer to use `Dor::WorkflowObject.initial_repo(name)` (where `name` is the name of the workflow we want to put in there, which is in our config as `Dor::Config.release.workflow_name`).  `Dor::Config.release.workflow_name` is also in the other parameter, as we always want to initialize that particular workflow (and thus this will be used to look up its xml)